### PR TITLE
ISPN-5149 Nested field is reported as non-indexed

### PR DESCRIPTION
**Pull request by anistor**
Jira: https://issues.jboss.org/browse/ISPN-5149
        https://issues.jboss.org/browse/ISPN-5161

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryWithProtostreamAnnotationsTest.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            .valueEquivalence(ByteArrayEquivalence.INSTANCE)
+            .indexing().index(Index.ALL)
+            .addProperty("default.directory_provider", "ram")
+            .addProperty("lucene_version", "LUCENE_CURRENT");
+
+      cacheManager = TestCacheManagerFactory.createCacheManager(builder);
+      cacheManager.defineConfiguration(TEST_CACHE_NAME, builder.build());
+      cache = cacheManager.getCache(TEST_CACHE_NAME);
+
+      hotRodServer = HotRodClientTestingUtil.startHotRodServer(cacheManager);
```
Comment (galderz) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryWithProtostreamAnnotationsTest.java]: In spite of using custom cache manager configuration, and custom remote cache client configuration, I'd rather see this class extend `org.infinispan.client.hotrod.test.SingleHotRodServerTest` than carry on duplicating code...

Comment (galderz) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryWithProtostreamAnnotationsTest.java]: In fact, would it make sense to have an abstract class that extends `org.infinispan.client.hotrod.test.SingleHotRodServerTest` and does a lot of the common set up, e.g. configure the cache manager accordingly... ?

**Comment by anistor** [comment]
@I've modified the test to extend from SingleHotRodServerTest.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/QueryFacadeImpl.java [hunk 2]
```diff
@@ -270,6 +265,10 @@ private FieldDescriptor getFieldDescriptor(Descriptor messageDescriptor, String
          if (fd == null) {
             throw new IllegalArgumentException("Unknown field " + name + " in type " + messageDescriptor.getFullName());
          }
+         IndexingMetadata indexingMetadata = messageDescriptor.getProcessedAnnotation(IndexingMetadata.INDEXED_ANNOTATION);
+         if (indexingMetadata != null && !indexingMetadata.isFieldIndexed(fd.getNumber())) {
+            throw new IllegalArgumentException("Field " + name + " from type " + messageDescriptor.getFullName() + " is not indexed");
```
Comment (tristantarrant) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/QueryFacadeImpl.java]: Please use the log interface for throwing these kind of exceptions

**Comment by anistor** [comment]
@tristantarrant Done.

**Comment by anistor** [comment]
None of the CI failures are related to the changes.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request closed without merging**